### DISCUSSION TITLE
SpinningBarrelSprite for RangedWeapon

### DISCUSSION THREAD

**Discussion by KB0MB4** [FakeFishGames/Barotrauma#12741] [Ideas and suggestions]
Hello, we already gave moving clockwise and back and forth. In game we have things like chaingun and some decorative items that use animation without changing a sprite texture, please make it so:
```xml
      <RailSprite/>
      <SpinningBarrelSprite/>
      <MoveSound/>
      <StartMoveSound/>
      <EndMoveSound/>
      <ChargeSound/>
```
would work not only for turrets but for RangedWeapon type too. I almost made a perfect minigun, just lack the spinning part.

https://github.com/Regalis11/Barotrauma/assets/20744282/ee4964de-1b7b-4aaf-bc55-1ae85e073375

**Comment by jigardi** [comment]
I like this suggestion a lot.  Have my upvote!
